lib/wallet/test/unit/Cardano/Wallet/DB/Store/Submissions/New/StoreSpec.hs

```diff
@@ -0,0 +1,63 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving #-}
+
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Cardano.Wallet.DB.Store.Submissions.New.StoreSpec ( spec ) where
+
+import Prelude
+
+import Cardano.DB.Sqlite
+    ( ForeignKeysSetting (ForeignKeysDisabled) )
+import Cardano.Wallet.Api.TypesSpec
+    ()
+import Cardano.Wallet.DB.Fixtures
+    ( WalletProperty, logScale, withDBInMemory, withInitializedWalletProp )
+import Cardano.Wallet.DB.Sqlite.Types
+    ( TxId (..) )
+import Cardano.Wallet.DB.Store.Submissions.New.Operations
+    ( DeltaTxSubmissions, mkStoreSubmissions )
+import Cardano.Wallet.Primitive.Types
+    ( SlotNo (..) )
+import Cardano.Wallet.Submissions.OperationsSpec
+    ( genOperationsDelta )
+import Cardano.Wallet.Submissions.Submissions
+    ( Submissions (..) )
+import Control.Monad
+    ( void )
+import Fmt
+    ( Buildable (..) )
+import System.Random
+    ( Random )
+import Test.DBVar
+    ( prop_StoreUpdates )
+import Test.Hspec
+    ( Spec, around, describe, it )
+import Test.QuickCheck
+    ( Arbitrary (..), property )
+
+spec :: Spec
+spec = do
+    around (withDBInMemory ForeignKeysDisabled) $ do
+        describe "submissions via API for a single wallet store" $ do
+            it "respects store laws"
+                $ property . prop_SingleWalletStoreLawsOperations
+
+instance Buildable DeltaTxSubmissions where
+    build = build . show
+
+deriving instance Random SlotNo
```

Comment:
Huh, where does `Random` come from? 🤔 We probably want an instance `Arbitrary SlotNo` here, which we can create using `Cardano.Wallet.Gen.genSlotNo`. The `import Cardano.Wallet.Api.TypesSpec ()` probably imports orphan instances — looks like trouble to me. 😅

Reply:
Went down into a 🐰-hole trying to remove `Random` by relying on `Enum` instead. Turns out that that `SlotNo` uses unsigned or larger integers than `Int`, and we run into conversion errors. `Random` is great.